bundles/org.openhab.binding.modbus.helioseasycontrols/src/main/java/org/openhab/binding/modbus/helioseasycontrols/internal/HeliosEasyControlsHandler.java

```diff
@@ -0,0 +1,577 @@
+                                });
+                    }, failureCallback -> {
+                        handleError("{} encountered error writing to device: {}",
+                                failureCallback.getCause().getMessage(), lock);
+                    });
+        }
+    }
+
+    /**
+     * Formats the provided date to a string in the device's configured date format
+     *
+     * @param date the date to be formatted
+     * @return a string in the device's configured date format
+     */
+    private String formatDate(String variableName, ZonedDateTime date) {
+        String y = Integer.toString(date.getYear());
+        String m = Integer.toString(date.getMonthValue());
+        if (m.length() == 1) {
+            m = "0" + m;
+        }
+        String d = Integer.toString(date.getDayOfMonth());
+        if (d.length() == 1) {
+            d = "0" + d;
+        }
+        if (variableName.equals(HeliosEasyControlsBindingConstants.DATE)) { // fixed format for writing the system date
+            return d + "." + m + "." + y;
+        } else {
+            switch (this.dateFormat) {
+                case 0: // dd.mm.yyyy
+                    return d + "." + m + "." + y;
+                case 1: // mm.dd.yyyy
+                    return m + "." + d + "." + y;
+                case 2: // yyyy.mm.dd
+                    return y + "." + m + "." + d;
+                default:
+                    return d + "." + m + "." + y;
+            }
+        }
+    }
+
+    /**
+     * Returns a DateTimeType object based on the provided String and the device's configured date format
+     *
+     * @param date The date string read from the device
+     * @return A DateTimeType object representing the date or time specified
+     */
+    private DateTimeType toDateTime(String date) {
+        String[] dateTimeParts = null;
+        String dateTime = date;
+        try { // date
+            dateTimeParts = date.split("\\."); // try to split date components
+            if (dateTimeParts.length == 1) { // time
+                return DateTimeType.valueOf(date);
+            } else { // date - we'll try the device's date format
+                switch (this.dateFormat) {
+                    case 0: // dd.mm.yyyy
+                        dateTime = dateTimeParts[2] + "-" + dateTimeParts[1] + "-" + dateTimeParts[0];
+                        break;
+                    case 1: // mm.dd.yyyy
+                        dateTime = dateTimeParts[2] + "-" + dateTimeParts[0] + "-" + dateTimeParts[1];
+                        break;
+                    case 2: // yyyy.mm.dd
+                        dateTime = dateTimeParts[0] + "-" + dateTimeParts[1] + "-" + dateTimeParts[2];
+                        break;
+                    default:
+                        dateTime = dateTimeParts[2] + "-" + dateTimeParts[1] + "-" + dateTimeParts[0];
+                        break;
+                }
+                return DateTimeType.valueOf(dateTime);

```

Comment:
Here dateTime would only contain a date, no time information. So I think it should be fine.
See above - the protocol spec splits up date and time into different variables... so I believe the current implementation should be fine.
It would be more elegant though if I created a method to set date, time and difference to UTC at the same time...

Reply:
see above, should be OK now :)